server/src/capture.rs

```diff
@@ -21,37 +21,103 @@
 // Standard library
 use indoc::indoc;
 use std::fs;
-use std::io;
+use std::io::{self, Write};
 use std::path::Path;
+use std::path::PathBuf;
 use std::sync::Arc;
 
 // Third-party
+use async_once_cell::OnceCell;
 use chrono::Local;
+use dirs::config_dir;
 use log::{error, info};
 use serde::{Deserialize, Serialize};
 use tokio::sync::Mutex;
 use tokio_postgres::{Client, NoTls};
+use uuid::Uuid;
+
+pub static GLOBAL_EVENT_CAPTURE: OnceCell<Arc<EventCapture>> = OnceCell::new();
+
+pub async fn get_event_capture() -> Arc<EventCapture> {
+    GLOBAL_EVENT_CAPTURE
+        .get_or_init(async {
+            let capture = EventCapture::new("config.json")
+                .await
+                .expect("Failed to initialize EventCapture");
+            Arc::new(capture)
+        })
+        .await
+        .clone()
+}
 
 // Local
 
+/// Returns the path where the identifier file is stored.
+fn get_uuid_file_path() -> Result<PathBuf, io::Error> {
+    let mut config_path = config_dir().ok_or_else(|| {
+        io::Error::new(
+            io::ErrorKind::NotFound,
+            "Could not find configuration directory",
+        )
+    })?;
+    config_path.push("vsc_extension");
+    config_path.push("uuid.txt");
+    Ok(config_path)
+}
+
+/// Creates a new UUID and writes it to the file if it does not already exist.
+fn create_uuid() -> Result<(), io::Error> {
+    let path = self::get_uuid_file_path()?;
+
+    // Check if the file already exists
+    if path.exists() {
+        return Ok(()); // UUID already exists, no need to create
+    }
+
+    // Generate a new UUID
+    let new_uuid = Uuid::new_v4().to_string();
+
+    // Ensure the parent directory exists
+    if let Some(parent) = path.parent() {
```

Comment:
Wrong logic -- only runs if `path.parent()` is Some, not if it exists. The parent should always exist. I'm guessing you want
``` Rust
    if !path.parent().expect("Config directory has no parent").exists()`.
```